chain/x/zoracle/client/cli/tx.go

```diff
@@ -118,3 +130,157 @@ func GetCmdReport(cdc *codec.Codec) *cobra.Command {
 		},
 	}
 }
+
+// GetCmdCreateDataSource implements the create data source command handler
+func GetCmdCreateDataSource(cdc *codec.Codec) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "create-data-source (--name [name]) (--script [path_to_script]) (--call-fee [fee]) (--owner [owner])",
+		Short: "Create new data source",
+		Args:  cobra.NoArgs,
+		Long: strings.TrimSpace(
+			fmt.Sprintf(`Create new data source that will be used by oracle script.
+Example:
+$ %s tx zoracle create-data-source --name coingecko-price --script ../price.sh --call-fee 100uband --owner band15d4apf20449ajvwycq8ruaypt7v6d345n9fpt9 --from mykey
+`,
+				version.ClientName,
+			),
+		),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			cliCtx := context.NewCLIContext().WithCodec(cdc)
+			txBldr := auth.NewTxBuilderFromCLI().WithTxEncoder(utils.GetTxEncoder(cdc))
+
+			name, err := cmd.Flags().GetString(flagName)
+			if err != nil {
+				return err
+			}
+			scriptPath, err := cmd.Flags().GetString(flagScript)
+			if err != nil {
+				return err
+			}
+			execBytes, err := readFile(scriptPath)
+			if err != nil {
+				return err
+			}
+
+			feeStr, err := cmd.Flags().GetString(flagCallFee)
+			if err != nil {
+				return err
+			}
+
+			fee, err := sdk.ParseCoins(feeStr)
+			if err != nil {
+				return err
+			}
+
+			ownerStr, err := cmd.Flags().GetString(flagOwner)
+			if err != nil {
+				return err
+			}
+			owner, err := sdk.AccAddressFromBech32(ownerStr)
+			if err != nil {
+				return err
+			}
+
+			msg := types.NewMsgCreateDataSource(
+				owner,
+				name,
+				fee,
+				execBytes,
+				cliCtx.GetFromAddress(),
+			)
+
+			err = msg.ValidateBasic()
+			if err != nil {
+				return err
+			}
+
+			return utils.GenerateOrBroadcastMsgs(cliCtx, txBldr, []sdk.Msg{msg})
+		},
+	}
+	cmd.Flags().String(flagName, "", "name of data source")
+	cmd.Flags().String(flagScript, "", "path to data source script")
+	cmd.Flags().String(flagCallFee, "", "fee for query this data source")
+	cmd.Flags().String(flagOwner, "", "owner of this data source")
+
+	return cmd
+}
+
+// GetCmdEditDataSource implements the edit data source command handler
+func GetCmdEditDataSource(cdc *codec.Codec) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "edit-data-source [id] (--name [name]) (--script [path_to_script]) (--call-fee [fee]) (--owner [owner])",
+		Short: "Edit data source",
+		Args:  cobra.ExactArgs(1),
+		Long: strings.TrimSpace(
+			fmt.Sprintf(`Edit existed data source by current owner.
```

Comment:
Edit an existing data source. The caller must be the current data source's owner.